app/src/main/java/tech/ula/utils/ValidationUtility.kt

```diff
@@ -1,31 +1,86 @@
 package tech.ula.utils
 
+import android.content.res.Resources
+import tech.ula.R
 import java.util.regex.Matcher
 import java.util.regex.Pattern
 
-class ValidationUtility() {
+class ValidationUtility(systemResources: Resources) {
 
-    fun isUsernameValid(password: String): Boolean {
-        val pattern: Pattern
-        val matcher: Matcher
+    private val resources = systemResources
 
-        val USERNAME_REGEX = "^[a-zA-Z0-9]*\$"
+    fun validateUsername(username: String): Credential {
+        var usernameIsValid = false
+        var errorMessage = ""
+        val blacklist = resources.getStringArray(R.array.blacklisted_usernames)
 
-        pattern = Pattern.compile(USERNAME_REGEX)
-        matcher = pattern.matcher(password)
+        when {
```

Comment:
I think you can do the same trick with the`when` expression here. Something like
```
return when {
    username.isEmpty -> {
        val errorMessage = resources.getString()
        Credential(false, errorMessage)
    }
...
}

Reply:
Alright I changed it here too.